measurements/pages/__init__.py

```diff
@@ -204,34 +208,69 @@ def files_in_country(country_code):
                            order_by=order_by,
                            current_country=country_code)
 
-
-@pages_blueprint.route('/files/download/<filename>')
-def files_download(filename):
-    try:
-        report = current_app.db_session.query(Report) \
-                        .filter(Report.textname.like("%" + filename)).first()
-        # XXX
-        # We have duplicate measurements :(
-        # So the below exception actually happens. This should be resolved
-        # in the data processing pipeline.
-        #   Report.filename == filename).one()
-    except NoResultFound:
-        raise NotFound("No file with that filename found")
-    except MultipleResultsFound:
-        # This should never happen.
-        raise HTTPException("Duplicate measurement detected")
-
-    if report is None:
+def decompress_autoclaved(autoclaved_filename, frames_off, frames_size, intra_off, intra_size):
+    def generator():
+        url = urljoin(current_app.config['AUTOCLAVED_BASE_URL'], autoclaved_filename)
+
+        current_app.logger.debug("Fetching autoclaved from: %s" % url)
+        current_app.logger.debug("  frames_off: %d" % frames_off)
+        current_app.logger.debug("  frames_size: %d" % frames_size)
+        current_app.logger.debug("  intra_off: %d" % intra_off)
+        current_app.logger.debug("  intra_size: %d" % intra_size)
+        try:
+            r = requests.get(url, headers={"Range": "bytes={}-{}".format(frames_off, frames_off + frames_size)},
+                                stream=True)
+            streamed_data = 0
+            for chunk in lz4framed.Decompressor(r.raw):
+                current_app.logger.debug("chunking away 1")
+                d = chunk
+                current_app.logger.debug("chunking away 2")
+                if streamed_data == 0 and intra_off > 0:
+                    d = chunk[intra_off:]
+                if streamed_data > (intra_size + len(d)):
+                    d = chunk[:(streamed_data - intra_size)]
```

Comment:
Not very clear check

Reply:
I more or less understand, but isn't it possible to express the final clamping in a more intuitive way?

Reply:
This is to handle the case in which we have streamed more data than what we needed (for example because the offset inside of the decompressed blob ends before the end of all the decompressed data).

Example say I wanted the decompressed bytes that go from 2 - 15

````
[ 0 1 2 3 4 5 6 7 8 9 10 ] <- compressed blob
````

becomes uncompressed:

```
[ 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20] <- uncompressed blob
```

I need to take the chunk that goes from 2-15

Reply:
Mumble:

```diff
+                if streamed_data > (intra_size + len(d)):
+                    d = chunk[:(streamed_data - intra_size)]
```

Re-reading this I am still in doubt.

What you want here, is to guarantee that `streamed_data + len(d) <= intra_size`, isn't it?

Reply:
I initially also find it confusing that you use both `chunk` and `d`. But actually I see how you need both of them because there is at least one case where you need to cut both at the beginning and at the end. 👍